queue_services/entity-filer/tests/unit/filing_processors/test_annual_report.py

```diff
@@ -14,36 +14,57 @@
 """The Test Suites to ensure that the worker is operating correctly."""
 import copy
 import datetime
+import pytest
 import random
 from unittest.mock import patch
 
 from freezegun import freeze_time
-from legal_api.models import Business, Filing
+from legal_api.models import BatchProcessing, Business, Filing
 from registry_schemas.example_data import ANNUAL_REPORT
 
 # from entity_filer.filing_processors.filing_components import create_party, create_role
 from entity_filer.filing_meta import FilingMeta
 from entity_filer.worker import process_filing
 from tests.unit import (
+    create_batch,
+    create_batch_processing,
     create_business,
     create_filing,
 )
+from tests import EPOCH_DATETIME
 
 
-def test_process_ar_filing(app, session):
+@pytest.mark.parametrize('test_name,flag_on,in_dissolution,eligibility,legal_type,', [
+    ('AR successfully', True, False, False, 'CP'),
+    ('Not withdrawn from the dissolution process', True, True, True, 'BC'),
+    ('Withdrawn from the dissolution process', True, True, False, 'CP'),
+    ('AR successfully when flag is off', False, True, False, 'CP')
```

Comment:
these two scenarios aren't actually possible for CPs.  CPs cannot have `in_dissolution` = True

Reply:
Updated